libs/SalesforceSDKCommon/SalesforceSDKCommon/Classes/Keychain/KeychainManager.swift

```diff
@@ -255,11 +331,165 @@ public class KeychainHelper: NSObject {
+            return KeychainResult(data: nil, status: errSecSuccess)
+        }
+        keychainAccessibleAttribute = accessibleAttribute
+        let query: [String: Any] = [
+            String(kSecClass): kSecClassGenericPassword,
+            String(kSecMatchLimit): kSecMatchLimitAll,
+            String(kSecReturnAttributes): kCFBooleanTrue!,
+            String(kSecAttrCreator): String(KeychainItemManager.tag)]
+        
+        let updateQuery: [String: Any] = [
+            String(kSecClass): kSecClassGenericPassword,
+            String(kSecAttrCreator): String(KeychainItemManager.tag)]
+        
+        var queryResult: AnyObject?
+        let status = SecItemCopyMatching(query as CFDictionary, &queryResult)
+       
+        if status == errSecItemNotFound {
+            SalesforceLogger.log(KeychainHelper.self,level: .debug, message: "Attempt to update accessibility attribute for mobilesdk items, no items found")
+            return KeychainResult(error: KeychainItemManager.mapError(from: status), status: status)
+        }
+        
+        if status != errSecSuccess {
+            SalesforceLogger.log(KeychainHelper.self,level: .error, message: "Attempt to update accessibility attribute for mobilesdk items failed!")
+            return KeychainResult(error: KeychainItemManager.mapError(from: status), status: status)
+        }
+        
+        
+        if (queryResult as? [[String : Any]]) != nil {
+            SalesforceLogger.log(KeychainHelper.self,level: .info, message: "Retrieved keychain items, will now update!")
+            let kAttributes: [String: Any] = [String(kSecAttrAccessible): keychainAccessibleAttribute]
+            let status = SecItemUpdate(updateQuery as CFDictionary,
+                                       kAttributes as CFDictionary)
+            
+            if status != errSecSuccess {
+                SalesforceLogger.log(KeychainHelper.self,level: .error, message: "Attempt to update accessibility attribute for mobilesdk items failed!")
+                return KeychainResult(error: KeychainItemManager.mapError(from: status), status: status)
+          
+            }
+            SalesforceLogger.log(KeychainHelper.self,level: .info, message: "Attempt to update accessibility attribute for mobilesdk items succeeded!")
+        }
+        
+        var queryUpdateResult: AnyObject?
+        let readStatus = SecItemCopyMatching(query as CFDictionary, &queryUpdateResult)
+        if readStatus != errSecSuccess {
+            SalesforceLogger.log(KeychainHelper.self,level: .error, message: "Attempt to update accessibility attribute for mobilesdk items failed!")
+            return KeychainResult(error: KeychainItemManager.mapError(from: readStatus), status: readStatus)
+        }
+        
+        return KeychainResult(data: nil, status: readStatus)
+    }
+    
+    //Pre 9.1 Upgrade Handling
+    internal class KeychainUpgradeManager {
+        let managedKeys = ["com.salesforce.security.passcode",
```

Comment:
Are those all the keys we managed? What about SmartStore?

Reply:
Seems like it @wmathurin